infrahub_sdk/template/__init__.py

```diff
@@ -0,0 +1,161 @@
+from __future__ import annotations
+
+import linecache
+from pathlib import Path
+from typing import Any, Callable, NoReturn
+
+import jinja2
+from jinja2 import meta
+from jinja2.sandbox import SandboxedEnvironment
+from netutils.utils import jinja2_convenience_function
+from rich.syntax import Syntax
+from rich.traceback import Traceback
+
+from .constants import CURRATED_NETUTILS_FILTERS
+from .exceptions import (
+    JinjaTemplateError,
+    JinjaTemplateNotFoundError,
+    JinjaTemplateOperationViolationError,
+    JinjaTemplateSyntaxError,
+    JinjaTemplateUndefinedError,
+)
+from .models import UndefinedJinja2Error
+
+netutils_filters = jinja2_convenience_function()
+
+
+class Jinja2Template:
+    def __init__(self, template: str | Path, template_directory: Path | None = None) -> None:
```

Comment:
I think it would be good to have a way to provide additional filters when we initialized this class, it would also make it easier to test it

Reply:
All good, I was just curious

Reply:
This has been added now.